ibm/service/power/data_source_ibm_pi_host.go

```diff
@@ -0,0 +1,201 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+)
+
+func DataSourceIBMPIHost() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIHostRead,
+
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_HostID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Host ID.",
```

Comment:
Could we get these properties sorted? Also for required arguments I've been putting ValidateFunc: validation.NoZeroValues. Should this be here as well?